Modules/FindCUDA/select_compute_arch.cmake

```diff
@@ -0,0 +1,175 @@
+# BorIFrowed from caffe
+# https://github.com/BVLC/caffe/blob/master/cmake/Cuda.cmake
+
+# Synopsis:
+# Function for selecting GPU arch flags for nvcc based on CUDA_ARCH_NAME
+# Usage:
+#   SELECT_NVCC_ARCH_FLAGS(out_variable)
+#
+# Variables affecting the choice:
+#
+# CUDA_ARCH_NAME: One of ("Fermi" "Kepler" "Kepler-M" "Maxwell" "Maxwell-M" "All" "Manual")
+# ENV{CUDA_ARCH_BIN} : Only tested if CUDA_ARCH_NAME set to "Manual"
+#
+
+# Known NVIDIA GPU achitectures
+# This list will be used for CUDA_ARCH_NAME = All option
+
+set(CUDA_KNOWN_GPU_ARCHITECTURES "2.0 2.1(2.0) 3.0 3.5 5.0")
+
+if (CUDA_VERSION VERSION_GREATER "6.5")
+  set(CUDA_KNOWN_GPU_ARCHITECTURES "3.2 3.7 ${CUDA_KNOWN_GPU_ARCHITECTURES} 5.2")
+  message("CUDA_VERSION VERSION_GREATER 6.5 YES ARCH=${CUDA_KNOWN_GPU_ARCHITECTURES}")
+endif ()
+
+if (CUDA_VERSION VERSION_GREATER "7.5")
+  set(CUDA_KNOWN_GPU_ARCHITECTURES "${CUDA_KNOWN_GPU_ARCHITECTURES} 6.0 6.2")
+  message("CUDA_VERSION VERSION_GREATER 7.5 YES ARCH=${CUDA_KNOWN_GPU_ARCHITECTURES}")
+endif ()
+
+
+################################################################################################
+# A function for automatic detection of GPUs installed  (IF autodetection is enabled)
+# Usage:
+#   CUDA_DETECT_INSTALLED_GPUS(OUT_VARIABLE)
+function(CUDA_DETECT_INSTALLED_GPUS OUT_VARIABLE)
```

Comment:
You should probably add CUDA_GPU_DETECT_OUTPUT to the list of cache variables removed when the version of the toolkit changes in FindCUDA.cmake (see cuda_unset_include_and_libraries).